tsconfig.strict.json

```diff
@@ -0,0 +1,16 @@
+{
+  "extends": "./tsconfig.base.json",
+  "compilerOptions": {
+    "exactOptionalPropertyTypes": true,
+    "isolatedModules": true,
+    "noPropertyAccessFromIndexSignature": true,
+    "noUncheckedIndexedAccess": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "strict": true,
+    // "verbatimModuleSyntax": true  // requires TypeScript 5
```

Comment:
Currently this cannot be enabled because the project is using TypeScript 4.5.

Maybe we can upgrade it in another PR?